src/modules/app/scanner.js

```diff
@@ -1,283 +1,43 @@
-        loadChildren({id: response.id, museumId, token, page: {
-          page: currentPage || 1,
-          limit: PER_PAGE
-        }});
-        clearBuffer();
-      });
-  } else if (nodePickList()) {
-    addNode({value: response, path: getPath(response)});
-    clearBuffer();
-  } else if (storageFacility()) {
-    goTo(`/magasin/${response.id}`);
-    clearBuffer();
-  } else {
-    error();
-  }
-};
-
-export const scanForNode = (
-  ajaxGet = ajax.simpleGet,
-  goTo =  hashHistory.push,
-  showError = emitError,
-  clearSearch = () => clearSearch$.next(),
-  clearBuffer = () => clearBuffer$.next(),
-  loadNode = (node) => loadNode$.next(node),
-  loadChildren = (node) => loadChildren$.next(node),
-  addNode = (node) => addNode$.next(node),
-  moveDialog = isMoveDialogActive,
-  nodePickList = isNodePickList,
-  storageFacility = isStorageFacility,
-  moveHistory = isMoveHistoryActive,
-  moveDialog$ = moveDialogStore$
-) => ({ uuid, museumId, token }) => {
-  MusitNode.findByUUID(ajaxGet)({uuid, museumId, token})
-    .do(response => {
-      if (response && response.nodeId) {
-        actOnNode(
-          response,
-          museumId,
-          token,
-          goTo,
-          showError,
-          clearSearch,
-          clearBuffer,
-          loadNode,
-          loadChildren,
-          addNode,
-          moveDialog,
-          nodePickList,
-          storageFacility,
-          moveHistory,
-          moveDialog$
-        );
-      } else {
-        showError({ message: I18n.t('musit.errorMainMessages.scanner.noMatchingNode', { uuid })});
-        clearSearch();
-      }
-    }).toPromise();
-};
-
-export const actOnObject = (
-  barcode,
-  response,
-  showError = emitError,
-  addMatches = (matches) => addMatches$.next(matches),
-  addObject = (object) => addObject$.next(object),
-  clearSearch = () => clearSearch$.next(),
-  clearBuffer = () => clearBuffer$.next(),
-  goTo = hashHistory.push,
-  objectPickList = isObjectPickList,
-  storageFacility = isStorageFacility,
-  moveDialog = isMoveDialogActive,
-  moveHistory = isMoveHistoryActive
-) => {
-  if (response.length === 1 || !Array.isArray(response)) {
-    const item = response.length ? response[0] : response;
-    const error = () => {
-      showError({ message: I18n.t('musit.errorMainMessages.scanner.cannotActOnObject')});
-      clearSearch();
-    };
-    if (moveHistory() || moveDialog()) {
-      error();
-    } else if (objectPickList()) {
-      addObject({ value: item, path: getPath(item)});
-      clearBuffer();
-    } else if (storageFacility()) {
-      if (!item.currentLocationId) {
-        showError({ message: I18n.t('musit.errorMainMessages.scanner.noCurrentLocation')});
-        clearSearch();
-      } else {
-        goTo(`/magasin/${item.currentLocationId}/objects`);
-        clearBuffer();
-      }
-    } else {
-      error();
+import omit from 'lodash/omit';
+
+const initialState = { buffer: '', code: '', uuid: false };
```

Comment:
Something we should consider but might not need to do now. 
The `initialState` can be mutated internally so it will be a good idea to change it to a function that return a fresh copy or introduce immutable js. From what I can see in the code we're not doing any mutation on the object itself here but it's something we should keep in mind.